src/wallet/coin_selection.rs

```diff
@@ -171,6 +145,178 @@ impl CoinSelectionResult {
     }
 }
 
+/// Perform the coin selection
+///
+/// - `algorithm`: the algorithm to use to select the UTXOs for the transaction
+/// - `database`: database reference to use by those algorithms that require extra information
+///               about UTXOs
+/// - `required_utxos`: the utxos that must be spent regardless of `target_amount` with their
+///                     weight cost
+/// - `optional_utxos`: the remaining available utxos to satisfy `target_amount` with their
+///                     weight cost
+/// - `fee_rate`: fee rate to use
+/// - `target_amount`: the outgoing amount in satoshis and the fees already
+///                    accumulated from added outputs and transaction’s header.
+/// - `weighted_drain_script`: the drain script used to create the change output associated with
+///                            the max satisfaction weight needed to spend it
+pub fn get_selection<D: Database, Cs: CoinSelectionAlgorithm<D>>(
```

Comment:
Mh, I'd like a more explicative name. What about `process_and_select_coins`?

Reply:
Sure, that's better. I can proceed with your suggested name and change it again in the future if we find a better one.